tgui/packages/tgui/interfaces/PlaneMasterDebug/index.tsx

```diff
@@ -0,0 +1,640 @@
+            )}
+            <Stack.Item>
+              <Button
+                color="transparent"
+                tooltip="Debugger Documentation"
+                icon="question"
+                selected={infoOpen}
+                onClick={() => setInfoOpen(true)}
+              />
+            </Stack.Item>
+            {!!(mob_ref !== our_ref) && (
+              <Stack.Item>
+                <Button
+                  color="transparent"
+                  tooltip="Reset Mob Focus"
+                  icon="magnifying-glass"
+                  onClick={() => act('reset_mob')}
+                />
+              </Stack.Item>
+            )}
+            <Stack.Item>
+              <Button
+                color="transparent"
+                tooltip="View Mirroring"
+                icon={our_ref !== mob_ref ? 'ghost' : 'eye'}
+                selected={tracking_active}
+                onClick={() => act('toggle_mirroring')}
+              />
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                color="transparent"
+                tooltip="View Mob Variables"
+                icon="pen"
+                onClick={() => act('vv_mob')}
+              />
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                color="transparent"
+                tooltip="Rebuild Plane Masters"
+                icon="recycle"
+                onClick={() => act('rebuild')}
+              />
+            </Stack.Item>
+          </Stack>
+        }
+      >
+        <Window.Content
+          style={{
+            backgroundImage: 'none',
+          }}
+        >
+          <InfinitePlane
+            width="100%"
+            height="100%"
+            backgroundImage={resolveAsset('grid_background.png')}
+            imageWidth={900}
+            initialLeft={500}
+            initialTop={-1350}
+          >
+            {planes.map((plane) => (
+              <PlaneMaster
+                key={plane.name}
+                plane={planesProcessed[plane.plane]}
+                connectionData={connectionDom.current}
+              />
+            ))}
+            <Connections connections={connections} />
+          </InfinitePlane>
+          {!!planeOpen && <PlaneEditor />}
+          <PlaneMenus />
+        </Window.Content>
+      </Window>
+    </PlaneDebugContext.Provider>
+  );
+}
+
+type Context = {
+  connectionHighlight: PlaneHighlight | undefined;
+  setConnectionHighlight: Dispatch<SetStateAction<PlaneHighlight | undefined>>;
+  activePlane: number | undefined;
+  setActivePlane: Dispatch<SetStateAction<number | undefined>>;
+  connectionOpen: boolean;
+  setConnectionOpen: Dispatch<SetStateAction<boolean>>;
+  infoOpen: boolean;
+  setInfoOpen: Dispatch<SetStateAction<boolean>>;
+  planeOpen: boolean;
+  setPlaneOpen: Dispatch<SetStateAction<boolean>>;
+  planesProcessed: PlaneMap;
+  act: Function;
+};
+
+export const PlaneDebugContext = createContext({} as Context);
```

Comment:
A bit of a style nit but this should be packaged in its own file and exported with a useful hook name, for example,

https://github.com/tgstation/tgstation/blob/master/tgui/packages/tgui/interfaces/PreferencesMenu/useServerPrefs.ts

You don't have to initialize it like it's done there